site/book/05-developing-functions/02-developing-in-Go.md

```diff
@@ -31,31 +31,36 @@ provided value:
 package main
 
 import (
-	"os"
+  "fmt"
+  "os"
 
-	"sigs.k8s.io/kustomize/kyaml/fn/framework"
-	"sigs.k8s.io/kustomize/kyaml/yaml"
+  "sigs.k8s.io/kustomize/kyaml/fn/framework"
+  "sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+  "sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
 var value string
 
 func main() {
-	resourceList := &framework.ResourceList{}
-	cmd := framework.Command(resourceList, func() error {
-		// cmd.Execute() will parse the ResourceList.functionConfig into
-		// cmd.Flags from the ResourceList.functionConfig.data field.
-		for i := range resourceList.Items {
-			// modify the resources using the kyaml/yaml library:
-			if err := resourceList.Items[i].PipeE(yaml.SetAnnotation("myannotation", value)); err != nil {
-				return err
-			}
-		}
-		return nil
-	})
-	cmd.Flags().StringVar(&value, "value", "", "annotation value")
-	if err := cmd.Execute(); err != nil {
-		os.Exit(1)
-	}
+  mp := MyProcessor{}
+  cmd := command.Build(&mp, command.StandaloneEnabled, false)
+  cmd.Flags().StringVar(&value, "value", "", "annotation value")
+  if err := cmd.Execute(); err != nil {
+    fmt.Fprintln(os.Stderr, err)
+    os.Exit(1)
+  }
+}
+
+type MyProcessor struct{}
```

Comment:
Drive by comment. It might useful to call out that `MyProcessor` is used to implement a `ResourceListProcessor`, so someone unfamiliar will understand why we need this setup.

Reply:
+1.
I would suggest to rename it to `Annotator`.